libs/h3/src/test/java/org/elasticsearch/h3/CellBoundaryTests.java

```diff
@@ -171,8 +174,53 @@ private void processOne(String h3Address, BufferedReader reader) throws IOExcept
         CellBoundary boundary = H3.h3ToGeoBoundary(h3Address);
         assert boundary.numPoints() == points.size();
         for (int i = 0; i < boundary.numPoints(); i++) {
-            assertEquals(h3Address, points.get(i)[0], boundary.getLatLon(i).getLatDeg(), 1e-8);
-            assertEquals(h3Address, points.get(i)[1], boundary.getLatLon(i).getLonDeg(), 1e-8);
+            assertEquals(h3Address, points.get(i)[0], boundary.getLatLon(i).getLatDeg(), 5e-7);
+            assertEquals(h3Address, points.get(i)[1], boundary.getLatLon(i).getLonDeg(), 5e-7);
         }
     }
+
+    public void testNumericEquivalentSharedBoundary() {
+        // we consider boundaries numerical equivalent if after encoded them using lucene, they resolve to the same number.
+        long h3 = H3.geoToH3(GeoTestUtil.nextLatitude(), GeoTestUtil.nextLongitude(), randomIntBetween(0, 15));
+        CellBoundary boundary = H3.h3ToGeoBoundary(h3);
+        for (long r : H3.hexRing(h3)) {
+            int count = 0;
+            CellBoundary ringBoundary = H3.h3ToGeoBoundary(r);
+            for (int i = 0; i < boundary.numPoints(); i++) {
+                LatLng latLng1 = boundary.getLatLon(getIndex(i, boundary.numPoints()));
+                LatLng latLng2 = boundary.getLatLon(getIndex(i + 1, boundary.numPoints()));
+                int lon1 = GeoEncodingUtils.encodeLongitude(latLng1.getLonDeg());
+                int lat1 = GeoEncodingUtils.encodeLatitude(latLng1.getLatDeg());
+                int lon2 = GeoEncodingUtils.encodeLongitude(latLng2.getLonDeg());
+                int lat2 = GeoEncodingUtils.encodeLatitude(latLng2.getLatDeg());
+                if (isSharedBoundary(lon1, lat1, lon2, lat2, ringBoundary)) {
+                    count++;
+                }
+            }
+            assertThat(count, Matchers.greaterThan(0));
```

Comment:
I ran this with `-Dtests.iters=10000`, and 233/10000 (2%) were `count==2` and 98% `count==1`. So perhaps we can assert also count not greater than 2.

Reply:
I figured out that the count is 1 for normal shaped cells, but when the cell crosses the edges of the icosohedron triangle, here are actually two shared edges, because the normal edge is broken into two at the triangle edge.